### PR TITLE
Update command clear

### PR DESCRIPTION
**Pull request by huanghao1998**
What's change
- now clear can take in tag predicate to clear all person associated to the tag
- add tests to the clear command

Model, ModelManager.java
- add hasDeleteMultiplePerson(): removes if the predicate satisfy; returns true if any person is removed, false otherwise

UniquePersonList.java
- add removeAll(): removes if the predicate satisfy

TagContainsKeywordPredicate.java
- add isEmpty(): check whether is the predicate empty

ClearCommand.java
- If the predicate is empty, clears all person
- else if the predicate does not contain in the existing tag, throw error
- if not, clear person associated to the tags (case insensitive)

ClearCommandParser.java
- Works similar to find command

### CODE REVIEW

File: src/main/java/seedu/socket/model/person/UniquePersonList.java [hunk 1]
```diff
@@ -98,6 +99,16 @@ public void remove(Person toRemove) {
...
     }
 
+    /**
+     * Removes all persons by the given {@code predicate}.
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    public boolean removeAll(Predicate<Person> predicate) {
+        requireNonNull(predicate);
+        boolean getRemoved = internalList.removeIf(predicate);
+        return getRemoved;
```
Comment (nus-jeff) [inline src/main/java/seedu/socket/model/person/UniquePersonList.java:109]: Minor nitpick from me, but maybe would be better to name it isRemoved to better reflect boolean field, but if you think it is ok to keep the current name, can just ignore this comment.

**Review by nus-jeff — COMMENTED** [review]
LGTM 👍 Just minor nitpicks, do take a look!

File: src/main/java/seedu/socket/model/person/UniquePersonList.java (hunk 1 above)
Comment (huanghao1998) [reply src/main/java/seedu/socket/model/person/UniquePersonList.java:109]: Oh i see, thanks for the suggestion, I would change it later

**Review by nus-jeff — APPROVED** [review]
LGTM 👍 Thanks~

**Pull request merged**